public/autoreload.html

```diff
@@ -0,0 +1,25 @@
+<script type="text/javascript">
+  // The etag value will be rendered on first page load
+  // by the server to store the timestamp of the last data update.
+  const etag = '{{ etag }}';
+
+  /**
+   * Periodically check the server whether we should reload the page
+   * since the content has been changed.
+   */
+  function checkReloadStatus() {
+    fetch('/api/reload', {
+      method: 'GET',
+      headers: { 'If-None-Match': etag },
+    }).then((response) => {
+      // The server will return `304 NOT MODIFIED` if the content of the Kedro project hasn't changed.
+      // Otherwise, it will return a `200` with an empty body.
+      // We should only reload on 200 and ignore all other status code like 500 as well.
+      if (response.status === 200) {
+        location.reload();
+      }
+    });
+  }
+
+  setInterval(checkReloadStatus, 3000);
```

Comment:
I wonder what is the decision to set this as 3 seconds? This is a really great use case to urge us to migrate to GraphQL for GraphQL subscription so we don't need to set up this type of `setInterval` to constantly ping the server :grin:

Reply:
@limdauto what was the thinking for the 3 secs?

Reply:
Yes, the proper way to do this is websocket. GraphQL subscription is managed websocket. But we don't want to bring another communication protocol into the mix just for this relatively minor feature. And this is how people have been doing live reload since the beginning on the web: https://livejs.com/

Re 3s: it was 1s in the beginning but that seems excessive. It takes at least a second for the backend to reload and read new content after a change. Testing it out shows 3s to be "instant" enough.

Reply:
@limdauto sure thing, I know that graphQL subscription is websocket and of course am not urging you to set up websocket / GraphQL just for this :grin:  - am just taking this as another chance to build up another case to nudge us for future migration to GraphQL ( which I understand is no small feat given the need to set up web socket to facilitate that.)